src/v1/datum.rs

```diff
@@ -8,11 +8,17 @@ use lbr_prelude::json::Json;
 use serde::{Deserialize, Serialize};
 
 /// blake2b-256 hash of a datum
-#[derive(Clone, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
 #[cfg_attr(feature = "lbf", derive(Json))]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub struct DatumHash(pub LedgerBytes);
 
+impl std::hash::Hash for DatumHash {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        state.write(&self.0 .0)
+    }
+}
```

Comment:
@szg251 did raise a good point, I don't think we should just pass the underlying Blake hash and just use `derive` on all of these as well. These serve a very different purpose imo.

Wdyt @chfanghr

Reply:
Well from the point of efficiency, it just feels weird to hash a hash.

Reply:
From the point of semantics imo it's weird to not hash a hash. I don't really know what 'Hash' assumes, so using deriving throughout makes sense. There's several questions that pop out to me. Is there some length that's expected? Maybe `Hash` is more lenient and works internally with an collision expectation?

All these questions I don't really want to care about, and rather just use their internal generic derivation, which completely abolishes asking these questions.

On the point of efficiency; I generally deprioritize efficiency when it comes to uniformity. Anything requiring ad-hoc consideration is causing maintenance friction, which we can't really afford and must priority minimizing such concerns.

I hope that makes sense

Reply:
Understood, and it's done.